sql/hive-thriftserver/pom.xml

```diff
@@ -41,11 +41,8 @@
       <version>${project.version}</version>
     </dependency>
     <dependency>
-      <groupId>org.apache.spark</groupId>
-      <artifactId>spark-core_${scala.binary.version}</artifactId>
-      <version>${project.version}</version>
-      <type>test-jar</type>
-      <scope>test</scope>
+      <groupId>org.eclipse.jetty</groupId>
+      <artifactId>jetty-servlet</artifactId>
```

Comment:
Ideally, moving the test scope jars to the testCompilation profile should not impact building main - unfortunately, it did. The error message I pasted above was the result of the build run.
Adding org.eclipse.jetty:jetty-servlet 'fixed' the issue - but I am not sure why this happened to begin with.

From dependency tree, I actually dont see a need for org.eclipse.* ... but I might be missing something.
Since you have more experience with build, any thoughts or suggestions @srowen ?

```
[INFO] ------------------------------------------------------------------------
[INFO] Building Spark Project Hive Thrift Server 2.1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- maven-dependency-plugin:2.10:tree (default-cli) @ spark-hive-thriftserver_2.11 ---
[INFO] org.apache.spark:spark-hive-thriftserver_2.11:jar:2.1.0-SNAPSHOT
[INFO] +- org.apache.spark:spark-hive_2.11:jar:2.1.0-SNAPSHOT:compile
[INFO] |  +- com.twitter:parquet-hadoop-bundle:jar:1.6.0:compile
[INFO] |  +- org.apache.spark:spark-core_2.11:jar:2.1.0-SNAPSHOT:compile
[INFO] |  +- org.apache.spark:spark-sql_2.11:jar:2.1.0-SNAPSHOT:compile
[INFO] |  +- org.spark-project.hive:hive-exec:jar:1.2.1.spark2:compile
[INFO] |  |  +- javolution:javolution:jar:5.5.1:compile
[INFO] |  |  +- log4j:apache-log4j-extras:jar:1.2.17:compile
[INFO] |  |  +- org.antlr:antlr-runtime:jar:3.4:compile
[INFO] |  |  |  +- org.antlr:stringtemplate:jar:3.2.1:compile
[INFO] |  |  |  \- antlr:antlr:jar:2.7.7:compile
[INFO] |  |  +- org.antlr:ST4:jar:4.0.4:compile
[INFO] |  |  +- org.apache.commons:commons-compress:jar:1.4.1:compile
[INFO] |  |  |  \- org.tukaani:xz:jar:1.0:compile
[INFO] |  |  +- org.codehaus.jackson:jackson-core-asl:jar:1.9.13:compile
[INFO] |  |  +- com.googlecode.javaewah:JavaEWAH:jar:0.3.2:compile
[INFO] |  |  +- org.iq80.snappy:snappy:jar:0.2:compile
[INFO] |  |  +- org.json:json:jar:20090211:compile
[INFO] |  |  +- stax:stax-api:jar:1.0.1:compile
[INFO] |  |  \- net.sf.opencsv:opencsv:jar:2.3:compile
[INFO] |  +- org.spark-project.hive:hive-metastore:jar:1.2.1.spark2:compile
[INFO] |  |  +- com.jolbox:bonecp:jar:0.8.0.RELEASE:compile
[INFO] |  |  +- org.apache.derby:derby:jar:10.12.1.1:compile
[INFO] |  |  +- org.datanucleus:datanucleus-api-jdo:jar:3.2.6:compile
[INFO] |  |  +- org.datanucleus:datanucleus-rdbms:jar:3.2.9:compile
[INFO] |  |  +- commons-pool:commons-pool:jar:1.5.4:compile
[INFO] |  |  +- commons-dbcp:commons-dbcp:jar:1.4:compile
[INFO] |  |  \- javax.jdo:jdo-api:jar:3.0.1:compile
[INFO] |  |     \- javax.transaction:jta:jar:1.1:compile
[INFO] |  +- org.apache.avro:avro:jar:1.7.7:compile
[INFO] |  |  \- com.thoughtworks.paranamer:paranamer:jar:2.3:compile
[INFO] |  +- org.apache.avro:avro-mapred:jar:hadoop2:1.7.7:compile
[INFO] |  |  +- org.apache.avro:avro-ipc:jar:1.7.7:compile
[INFO] |  |  \- org.apache.avro:avro-ipc:jar:tests:1.7.7:test
[INFO] |  +- commons-httpclient:commons-httpclient:jar:3.1:compile
[INFO] |  +- org.apache.calcite:calcite-avatica:jar:1.2.0-incubating:compile
[INFO] |  +- org.apache.calcite:calcite-core:jar:1.2.0-incubating:compile
[INFO] |  |  +- org.apache.calcite:calcite-linq4j:jar:1.2.0-incubating:compile
[INFO] |  |  +- net.hydromatic:eigenbase-properties:jar:1.1.5:compile
[INFO] |  |  \- org.codehaus.janino:commons-compiler:jar:2.7.6:compile
[INFO] |  +- org.apache.httpcomponents:httpclient:jar:4.5.2:compile
[INFO] |  +- org.codehaus.jackson:jackson-mapper-asl:jar:1.9.13:compile
[INFO] |  +- commons-codec:commons-codec:jar:1.10:compile
[INFO] |  +- joda-time:joda-time:jar:2.9.3:compile
[INFO] |  +- org.jodd:jodd-core:jar:3.5.2:compile
[INFO] |  +- com.google.code.findbugs:jsr305:jar:1.3.9:compile
[INFO] |  +- org.datanucleus:datanucleus-core:jar:3.2.10:compile
[INFO] |  +- org.apache.thrift:libthrift:jar:0.9.2:compile
[INFO] |  \- org.apache.thrift:libfb303:jar:0.9.2:compile
[INFO] +- org.apache.spark:spark-core_2.11:test-jar:tests:2.1.0-SNAPSHOT:test
[INFO] |  +- com.twitter:chill_2.11:jar:0.8.0:compile
[INFO] |  |  \- com.esotericsoftware:kryo-shaded:jar:3.0.3:compile
[INFO] |  |     +- com.esotericsoftware:minlog:jar:1.3.0:compile
[INFO] |  |     \- org.objenesis:objenesis:jar:2.1:compile
[INFO] |  +- com.twitter:chill-java:jar:0.8.0:compile
[INFO] |  +- org.apache.xbean:xbean-asm5-shaded:jar:4.4:compile
[INFO] |  +- org.apache.hadoop:hadoop-client:jar:2.7.3:compile
[INFO] |  |  +- org.apache.hadoop:hadoop-common:jar:2.7.3:compile
[INFO] |  |  |  +- xmlenc:xmlenc:jar:0.52:compile
[INFO] |  |  |  +- javax.servlet.jsp:jsp-api:jar:2.1:runtime
[INFO] |  |  |  +- commons-configuration:commons-configuration:jar:1.6:compile
[INFO] |  |  |  |  +- commons-digester:commons-digester:jar:1.8:compile
[INFO] |  |  |  |  |  \- commons-beanutils:commons-beanutils:jar:1.7.0:compile
[INFO] |  |  |  |  \- commons-beanutils:commons-beanutils-core:jar:1.8.0:compile
[INFO] |  |  |  +- com.google.protobuf:protobuf-java:jar:2.5.0:compile
[INFO] |  |  |  +- com.google.code.gson:gson:jar:2.2.4:compile
[INFO] |  |  |  +- org.apache.hadoop:hadoop-auth:jar:2.7.3:compile
[INFO] |  |  |  |  \- org.apache.directory.server:apacheds-kerberos-codec:jar:2.0.0-M15:compile
[INFO] |  |  |  |     +- org.apache.directory.server:apacheds-i18n:jar:2.0.0-M15:compile
[INFO] |  |  |  |     +- org.apache.directory.api:api-asn1-api:jar:1.0.0-M20:compile
[INFO] |  |  |  |     \- org.apache.directory.api:api-util:jar:1.0.0-M20:compile
[INFO] |  |  |  +- org.apache.curator:curator-client:jar:2.6.0:compile
[INFO] |  |  |  \- org.apache.htrace:htrace-core:jar:3.1.0-incubating:compile
[INFO] |  |  +- org.apache.hadoop:hadoop-hdfs:jar:2.7.3:compile
[INFO] |  |  |  \- org.mortbay.jetty:jetty-util:jar:6.1.26:compile
[INFO] |  |  +- org.apache.hadoop:hadoop-mapreduce-client-app:jar:2.7.3:compile
[INFO] |  |  |  +- org.apache.hadoop:hadoop-mapreduce-client-common:jar:2.7.3:compile
[INFO] |  |  |  |  +- org.apache.hadoop:hadoop-yarn-client:jar:2.7.3:compile
[INFO] |  |  |  |  \- org.apache.hadoop:hadoop-yarn-server-common:jar:2.7.3:compile
[INFO] |  |  |  \- org.apache.hadoop:hadoop-mapreduce-client-shuffle:jar:2.7.3:compile
[INFO] |  |  +- org.apache.hadoop:hadoop-yarn-api:jar:2.7.3:compile
[INFO] |  |  +- org.apache.hadoop:hadoop-mapreduce-client-core:jar:2.7.3:compile
[INFO] |  |  |  \- org.apache.hadoop:hadoop-yarn-common:jar:2.7.3:compile
[INFO] |  |  |     +- javax.xml.bind:jaxb-api:jar:2.2.2:compile
[INFO] |  |  |     |  \- javax.xml.stream:stax-api:jar:1.0-2:compile
[INFO] |  |  |     +- org.codehaus.jackson:jackson-jaxrs:jar:1.9.13:compile
[INFO] |  |  |     \- org.codehaus.jackson:jackson-xc:jar:1.9.13:compile
[INFO] |  |  +- org.apache.hadoop:hadoop-mapreduce-client-jobclient:jar:2.7.3:compile
[INFO] |  |  \- org.apache.hadoop:hadoop-annotations:jar:2.7.3:compile
[INFO] |  +- org.apache.spark:spark-launcher_2.11:jar:2.1.0-SNAPSHOT:compile
[INFO] |  +- org.apache.spark:spark-network-common_2.11:jar:2.1.0-SNAPSHOT:compile
[INFO] |  |  +- org.fusesource.leveldbjni:leveldbjni-all:jar:1.8:compile
[INFO] |  |  \- com.fasterxml.jackson.core:jackson-annotations:jar:2.6.5:compile
[INFO] |  +- org.apache.spark:spark-network-shuffle_2.11:jar:2.1.0-SNAPSHOT:compile
[INFO] |  +- org.apache.spark:spark-unsafe_2.11:jar:2.1.0-SNAPSHOT:compile
[INFO] |  +- net.java.dev.jets3t:jets3t:jar:0.9.3:compile
[INFO] |  |  +- javax.activation:activation:jar:1.1.1:compile
[INFO] |  |  +- mx4j:mx4j:jar:3.0.2:compile
[INFO] |  |  +- javax.mail:mail:jar:1.4.7:compile
[INFO] |  |  +- org.bouncycastle:bcprov-jdk15on:jar:1.51:compile
[INFO] |  |  \- com.jamesmurty.utils:java-xmlbuilder:jar:1.0:compile
[INFO] |  |     \- net.iharder:base64:jar:2.3.8:compile
[INFO] |  +- org.apache.curator:curator-recipes:jar:2.6.0:compile
[INFO] |  |  +- org.apache.curator:curator-framework:jar:2.6.0:compile
[INFO] |  |  \- org.apache.zookeeper:zookeeper:jar:3.4.6:compile
[INFO] |  +- org.eclipse.jetty:jetty-plus:jar:9.2.16.v20160414:provided
[INFO] |  |  +- org.eclipse.jetty:jetty-webapp:jar:9.2.16.v20160414:provided
[INFO] |  |  |  \- org.eclipse.jetty:jetty-xml:jar:9.2.16.v20160414:provided
[INFO] |  |  \- org.eclipse.jetty:jetty-jndi:jar:9.2.16.v20160414:provided
[INFO] |  +- org.eclipse.jetty:jetty-security:jar:9.2.16.v20160414:provided
[INFO] |  +- org.eclipse.jetty:jetty-util:jar:9.2.16.v20160414:provided
[INFO] |  +- org.eclipse.jetty:jetty-server:jar:9.2.16.v20160414:provided
[INFO] |  |  \- org.eclipse.jetty:jetty-io:jar:9.2.16.v20160414:provided
[INFO] |  +- org.eclipse.jetty:jetty-http:jar:9.2.16.v20160414:provided
[INFO] |  +- org.eclipse.jetty:jetty-continuation:jar:9.2.16.v20160414:provided
[INFO] |  +- org.eclipse.jetty:jetty-servlet:jar:9.2.16.v20160414:provided
[INFO] |  +- org.eclipse.jetty:jetty-proxy:jar:9.2.16.v20160414:provided
[INFO] |  +- org.eclipse.jetty:jetty-client:jar:9.2.16.v20160414:provided
[INFO] |  +- org.eclipse.jetty:jetty-servlets:jar:9.2.16.v20160414:provided
[INFO] |  +- javax.servlet:javax.servlet-api:jar:3.1.0:compile
[INFO] |  +- org.apache.commons:commons-lang3:jar:3.5:compile
[INFO] |  +- org.apache.commons:commons-math3:jar:3.4.1:compile
[INFO] |  +- org.slf4j:slf4j-api:jar:1.7.16:compile
[INFO] |  +- org.slf4j:jul-to-slf4j:jar:1.7.16:compile
[INFO] |  +- org.slf4j:jcl-over-slf4j:jar:1.7.16:compile
[INFO] |  +- log4j:log4j:jar:1.2.17:compile
[INFO] |  +- org.slf4j:slf4j-log4j12:jar:1.7.16:compile
[INFO] |  +- com.ning:compress-lzf:jar:1.0.3:compile
[INFO] |  +- org.xerial.snappy:snappy-java:jar:1.1.2.6:compile
[INFO] |  +- net.jpountz.lz4:lz4:jar:1.3.0:compile
[INFO] |  +- org.roaringbitmap:RoaringBitmap:jar:0.5.11:compile
[INFO] |  +- commons-net:commons-net:jar:2.2:compile
[INFO] |  +- org.scala-lang:scala-library:jar:2.11.8:compile
[INFO] |  +- org.json4s:json4s-jackson_2.11:jar:3.2.11:compile
[INFO] |  |  \- org.json4s:json4s-core_2.11:jar:3.2.11:compile
[INFO] |  |     +- org.json4s:json4s-ast_2.11:jar:3.2.11:compile
[INFO] |  |     \- org.scala-lang:scalap:jar:2.11.8:compile
[INFO] |  |        \- org.scala-lang:scala-compiler:jar:2.11.8:compile
[INFO] |  |           \- org.scala-lang.modules:scala-parser-combinators_2.11:jar:1.0.4:compile
[INFO] |  +- org.glassfish.jersey.core:jersey-client:jar:2.22.2:compile
[INFO] |  |  +- javax.ws.rs:javax.ws.rs-api:jar:2.0.1:compile
[INFO] |  |  +- org.glassfish.hk2:hk2-api:jar:2.4.0-b34:compile
[INFO] |  |  |  +- org.glassfish.hk2:hk2-utils:jar:2.4.0-b34:compile
[INFO] |  |  |  \- org.glassfish.hk2.external:aopalliance-repackaged:jar:2.4.0-b34:compile
[INFO] |  |  +- org.glassfish.hk2.external:javax.inject:jar:2.4.0-b34:compile
[INFO] |  |  \- org.glassfish.hk2:hk2-locator:jar:2.4.0-b34:compile
[INFO] |  |     \- org.javassist:javassist:jar:3.18.1-GA:compile
[INFO] |  +- org.glassfish.jersey.core:jersey-common:jar:2.22.2:compile
[INFO] |  |  +- javax.annotation:javax.annotation-api:jar:1.2:compile
[INFO] |  |  +- org.glassfish.jersey.bundles.repackaged:jersey-guava:jar:2.22.2:compile
[INFO] |  |  \- org.glassfish.hk2:osgi-resource-locator:jar:1.0.1:compile
[INFO] |  +- org.glassfish.jersey.core:jersey-server:jar:2.22.2:compile
[INFO] |  |  +- org.glassfish.jersey.media:jersey-media-jaxb:jar:2.22.2:compile
[INFO] |  |  \- javax.validation:validation-api:jar:1.1.0.Final:compile
[INFO] |  +- org.glassfish.jersey.containers:jersey-container-servlet:jar:2.22.2:compile
[INFO] |  +- org.glassfish.jersey.containers:jersey-container-servlet-core:jar:2.22.2:compile
[INFO] |  +- io.netty:netty-all:jar:4.0.41.Final:compile
[INFO] |  +- io.netty:netty:jar:3.8.0.Final:compile
[INFO] |  +- com.clearspring.analytics:stream:jar:2.7.0:compile
[INFO] |  +- io.dropwizard.metrics:metrics-core:jar:3.1.2:compile
[INFO] |  +- io.dropwizard.metrics:metrics-jvm:jar:3.1.2:compile
[INFO] |  +- io.dropwizard.metrics:metrics-json:jar:3.1.2:compile
[INFO] |  +- io.dropwizard.metrics:metrics-graphite:jar:3.1.2:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.6.5:compile
[INFO] |  |  \- com.fasterxml.jackson.core:jackson-core:jar:2.6.5:compile
[INFO] |  +- com.fasterxml.jackson.module:jackson-module-scala_2.11:jar:2.6.5:compile
[INFO] |  |  \- com.fasterxml.jackson.module:jackson-module-paranamer:jar:2.6.5:compile
[INFO] |  +- org.apache.ivy:ivy:jar:2.4.0:compile
[INFO] |  +- oro:oro:jar:2.0.8:compile
[INFO] |  +- net.razorvine:pyrolite:jar:4.13:compile
[INFO] |  +- net.sf.py4j:py4j:jar:0.10.3:compile
[INFO] |  \- org.apache.commons:commons-crypto:jar:1.0.0:compile
[INFO] +- com.google.guava:guava:jar:14.0.1:provided
[INFO] +- org.spark-project.hive:hive-cli:jar:1.2.1.spark2:compile
[INFO] |  +- commons-cli:commons-cli:jar:1.2:compile
[INFO] |  +- commons-lang:commons-lang:jar:2.6:compile
[INFO] |  \- jline:jline:jar:2.12.1:compile
[INFO] +- org.spark-project.hive:hive-jdbc:jar:1.2.1.spark2:compile
[INFO] |  \- org.apache.httpcomponents:httpcore:jar:4.4.4:compile
[INFO] +- org.spark-project.hive:hive-beeline:jar:1.2.1.spark2:compile
[INFO] |  +- commons-io:commons-io:jar:2.4:compile
[INFO] |  \- net.sf.supercsv:super-csv:jar:2.2.0:compile
[INFO] +- org.seleniumhq.selenium:selenium-java:jar:2.52.0:test
[INFO] |  +- org.seleniumhq.selenium:selenium-chrome-driver:jar:2.52.0:test
[INFO] |  |  \- org.seleniumhq.selenium:selenium-remote-driver:jar:2.52.0:test
[INFO] |  |     +- cglib:cglib-nodep:jar:2.1_3:test
[INFO] |  |     \- org.seleniumhq.selenium:selenium-api:jar:2.52.0:test
[INFO] |  +- org.seleniumhq.selenium:selenium-edge-driver:jar:2.52.0:test
[INFO] |  |  \- org.apache.commons:commons-exec:jar:1.3:test
[INFO] |  +- org.seleniumhq.selenium:selenium-firefox-driver:jar:2.52.0:test
[INFO] |  +- org.seleniumhq.selenium:selenium-ie-driver:jar:2.52.0:test
[INFO] |  |  +- net.java.dev.jna:jna:jar:4.1.0:test
[INFO] |  |  \- net.java.dev.jna:jna-platform:jar:4.1.0:test
[INFO] |  +- org.seleniumhq.selenium:selenium-safari-driver:jar:2.52.0:test
[INFO] |  +- org.seleniumhq.selenium:selenium-support:jar:2.52.0:test
[INFO] |  +- org.webbitserver:webbit:jar:0.4.14:test
[INFO] |  \- org.seleniumhq.selenium:selenium-leg-rc:jar:2.52.0:test
[INFO] +- org.seleniumhq.selenium:selenium-htmlunit-driver:jar:2.52.0:test
[INFO] |  +- net.sourceforge.htmlunit:htmlunit:jar:2.18:test
[INFO] |  |  +- xalan:xalan:jar:2.7.2:test
[INFO] |  |  |  \- xalan:serializer:jar:2.7.2:test
[INFO] |  |  +- org.apache.httpcomponents:httpmime:jar:4.5.2:test
[INFO] |  |  +- net.sourceforge.htmlunit:htmlunit-core-js:jar:2.17:test
[INFO] |  |  +- xerces:xercesImpl:jar:2.11.0:compile
[INFO] |  |  |  \- xml-apis:xml-apis:jar:1.4.01:test
[INFO] |  |  +- net.sourceforge.nekohtml:nekohtml:jar:1.9.22:test
[INFO] |  |  +- net.sourceforge.cssparser:cssparser:jar:0.9.16:test
[INFO] |  |  |  \- org.w3c.css:sac:jar:1.3:test
[INFO] |  |  \- org.eclipse.jetty.websocket:websocket-client:jar:9.2.12.v20150709:test
[INFO] |  |     \- org.eclipse.jetty.websocket:websocket-common:jar:9.2.12.v20150709:test
[INFO] |  |        \- org.eclipse.jetty.websocket:websocket-api:jar:9.2.12.v20150709:test
[INFO] |  \- commons-collections:commons-collections:jar:3.2.2:compile
[INFO] +- org.apache.spark:spark-sql_2.11:test-jar:tests:2.1.0-SNAPSHOT:test
[INFO] |  +- com.univocity:univocity-parsers:jar:2.2.1:compile
[INFO] |  +- org.apache.spark:spark-sketch_2.11:jar:2.1.0-SNAPSHOT:compile
[INFO] |  +- org.apache.spark:spark-catalyst_2.11:jar:2.1.0-SNAPSHOT:compile
[INFO] |  |  +- org.codehaus.janino:janino:jar:3.0.0:compile
[INFO] |  |  \- org.antlr:antlr4-runtime:jar:4.5.3:compile
[INFO] |  +- org.apache.parquet:parquet-column:jar:1.8.1:compile
[INFO] |  |  +- org.apache.parquet:parquet-common:jar:1.8.1:compile
[INFO] |  |  \- org.apache.parquet:parquet-encoding:jar:1.8.1:compile
[INFO] |  \- org.apache.parquet:parquet-hadoop:jar:1.8.1:compile
[INFO] |     +- org.apache.parquet:parquet-format:jar:2.3.0-incubating:compile
[INFO] |     \- org.apache.parquet:parquet-jackson:jar:1.8.1:compile
[INFO] +- org.apache.spark:spark-tags_2.11:jar:2.1.0-SNAPSHOT:compile
[INFO] +- net.sf.jpam:jpam:jar:1.1:compile
[INFO] |  \- commons-logging:commons-logging:jar:1.0.4:compile
[INFO] +- org.spark-project.spark:unused:jar:1.0.0:compile
[INFO] +- org.scalatest:scalatest_2.11:jar:2.2.6:test
[INFO] |  +- org.scala-lang:scala-reflect:jar:2.11.8:compile
[INFO] |  \- org.scala-lang.modules:scala-xml_2.11:jar:1.0.2:compile
[INFO] +- junit:junit:jar:4.12:test
[INFO] |  \- org.hamcrest:hamcrest-core:jar:1.3:test
[INFO] \- com.novocode:junit-interface:jar:0.11:test
[INFO]    \- org.scala-sbt:test-interface:jar:1.0:test
[INFO]
```